src/main/scala/com/microsoft/hyperspace/util/ResolverUtils.scala

```diff
@@ -71,4 +76,68 @@ object ResolverUtils {
       availableStrings: Seq[String]): Option[Seq[String]] = {
     Some(requiredStrings.map(resolve(spark, _, availableStrings).getOrElse { return None }))
   }
+
+  /**
+   * Finds all resolved strings for requiredStrings, from the given logical plan. Returns
+   * optional seq of resolved strings if all required strings are resolved, otherwise None.
+   *
+   * @param spark Spark session.
+   * @param requiredStrings List of strings to resolve.
+   * @param plan Logical plan to resolve against.
+   * @return Optional Seq of resolved strings if all required strings are resolved. Else, None.
+   */
+  def resolve(
+      spark: SparkSession,
+      requiredStrings: Seq[String],
+      plan: LogicalPlan): Option[Seq[String]] = {
+
+    def fixParts(parts: Seq[String], outputs: Seq[Attribute]): Seq[String] = {
+      var newParts = Seq.empty[String]
+      val h :: t = parts.toList
+      val topLevelField = outputs.find(_.name.compareToIgnoreCase(h) == 0)
+      topLevelField match {
+        case Some(o) =>
+          newParts = newParts :+ o.name
+          var children = o.dataType.asInstanceOf[StructType]
+          t.foreach { e =>
+            val elIdx: Int = children.fieldNames.indexWhere { f => f.compareToIgnoreCase(e) == 0 }
+            if (elIdx >= 0) {
+              newParts = newParts :+ children.fieldNames(elIdx)
+              children.fields.toSeq(elIdx).dataType match {
+                case s: StructType =>
+                  children = s
+                case _ =>
+              }
+            }
+          }
+        case None =>
+          newParts = h +: t
```

Comment:
is this a possible case?

Reply:
`find` method returns an `Option` so both `Some` and `None` are possible values and the correct matching would be to match against each possibility. 

Now, if you ask me how can we land in this case when a lot of other checks already did take place, the answer is I don't know. Still it is a best practice to cover both cases.

Reply:
I think early failure is better when something goes wrong. WDYT?

Reply:
Just falling back to original access path instead of the case fixed ones should be totally ok because the parts are already resolved and found by `resolveQuoted` and `.sql` methods. I would say that in case of any unknown issue in this resolving process it is always possible to go on with already resolved parts.

Given a dataset with the following nested field `struct("my", struct("nested", field("path")))` and `spark.sql.caseSensitive -> false`, using the following access field path `my.neSTed.pAth` it will get resolved to `my.neSTed.pAth` and the `.sql` method with return `` `my`.`neSTed`.`pAth` ``. Now, to comply with the tests you have put in place I am forced to create this `fixParts` method to turn the `Seq("my", "neSTed", "pAth")` into `Seq("my", "nested", "pAth")` and I guess this is due to the fact that we want to create the index dataset with the correct column name otherwise the index will be unusable if the user changes the `spark.sql.caseSensitive` to `true`.

So, are you suggesting to throw a `HyperspaceException` or just `get` and let it throw itself a 
`NoSuchElementException`?